### PR TITLE
detect when user changes web3 network or account

### LINKED ISSUE #2884: auto-login user when changing MM account
when user is logged into MM account and change MM account auto logout and back in to new MM chosen account. MM === meta mask

### PR DESCRIPTION
**Pull request by bthaile**
fixes https://github.com/AugurProject/augur/issues/2884

auto logout user if web3 address changes
detect if network has changed by web3 provider and show network mismatch modal.

### CODE REVIEW

File: packages/augur-ui/src/services/augursdk.ts [hunk 1]
```diff
@@ -30,6 +31,7 @@ export class SDK {
   ) {
     this.isWeb3Transport = isWeb3;
     this.env = env;
+    this.account = account;
```
Comment (bthaile) [inline packages/augur-ui/src/services/augursdk.ts:34]: used for comparisons like networkId

File: packages/augur-ui/src/modules/contracts/actions/contractCalls.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ import { NewMarket } from 'modules/types';
 import { TestNetReputationToken } from '@augurproject/core/build/libraries/GenericContractInterfaces';
 
 export function clearUserTx(): void {
-  const Augur = augurSdk.get();
+  // const Augur = augurSdk.get();
```
Comment (bthaile) [inline packages/augur-ui/src/modules/contracts/actions/contractCalls.ts:30]: this was causing issues when trying to switch accounts, augurSdk wasn't initialized.

File: packages/augur-ui/src/modules/alerts/selectors/alerts.ts [hunk 3]
```diff
@@ -13,8 +13,8 @@ export const selectInfoAlerts = selectAlertsByLevel(alertLevels.INFO);
 export const selectInfoAlertsAndSeenCount = createSelector(
   selectInfoAlerts,
   alerts => {
-    const { universe, connection } = store.getState();
-    if (!connection.isConnected) return { unseenCount: 0, alerts: [] };
+    const { universe, connection, authStatus } = store.getState();
+    if (!connection.isConnected || !authStatus.isLogged) return { unseenCount: 0, alerts: [] };
```
Comment (bthaile) [inline packages/augur-ui/src/modules/alerts/selectors/alerts.ts:17]: user need to be logged in to get alerts

**Review by andrewdamelio — APPROVED** [review]
nice!

**Pull request merged**